### ISSUE TITLE
Met helios compiling issue following time lock example

### ISSUE THREAD

**Issue by baibinghere** [Hyperion-BT/Helios#37]
I'm following [Time Lock](https://www.hyperion-bt.org/Helios-Book/cli/example-time-lock.html) example to compile time_lock.hl to JSON.

I met below issue:

```shell
root@VM-0-3-ubuntu:~/cardano/demos/time-lock# node --version
v16.15.1
# OS version is Ubuntu 22.04 LTS.
root@VM-0-3-ubuntu:~/cardano/demos/time-lock# helios compile time_lock.hl > time_lock.json
Error in /root/cardano/demos/time-lock/time_lock.hl
SyntaxError on line 21: invalid source character '$' (utf-8 not yet supported outside string literals)
```

The source code is exactly same as the example. (Except for the PubkeyHash key has been change to my own verification key.

```javascript
spending time_lock

struct Datum {
    lockUntil: Time
    owner:     PubKeyHash // can't get this info from the ScriptContext
    nonce:     Int // doesn't actually need be checked here
}

func main(datum: Datum, ctx: ScriptContext) -> Bool {
    tx: Tx = ctx.tx;
    now: Time = tx.time_range.start;
    returnToOwner: Bool = tx.is_signed_by(datum.owner);

    print("now: " + now.show() + ", lock: " + datum.lockUntil.show()); now > datum.lockUntil ||
    (print("returning? " + returnToOwner.show()); returnToOwner)
}
// end-of-main, anything that comes after isn't part of the on-chain script

// Helios can evaluate MY_DATUM into a data-structure that can be used to build a transaction
const MY_DATUM = Datum{
  lockUntil: Time::new(${(new Date()).getTime() + 1000*60*5}),
  owner: PubKeyHash::new(<My own verification key>),
  nonce: 42
}
```

**Comment by christianschmitz** [comment]
The issue here is that `Time::new(${...})` was written for inline javascript.

If you are just interested in the on-chain part then you can remove the `const MY_DATUM = ... ` statement

If however you want to use `MY_DATUM` to actually generate the datum data-structure using Helios: I haven't yet made that available in the cli, but I should be able to implement that soon.

**Comment by baibinghere** [comment]
Hi @christianschmitz,

Do you mean, inline javascript is not implemented yet? Since the example demonstrated this, I suppose it's already supported.

If `const MY_DATUM = ...` is removed, how could we execute below command.

```shell
helios eval time_lock.hl MY_DATUM
```

And looks like we won't have a lock time set right?

**Comment by christianschmitz** [comment]
I've removed the inline Javascript from the time-lock example Helios script.

Now the LOCK_UNTIL parameter is set using the cli. The other parameters can also be set using the cli. (The `-D` option is available since cli version >= v0.1.14).

For example: 
```
helios eval time_lock.hl MY_DATUM \
  -DOWNER "000102030405060708090a0b0c0d0e0f101112131415161718191a1b" \
  -DLOCK_UNTIL $(($(date +%s) + 300)) \
  -DNONCE 12345
  ```

Sorry for the confusion about inline javascript, Helios is a distinct language and doesn't support. What I meant by that is you can generate the Helios source code inside javascript.

Thank you for pointing this mistake out! If you find anything else please let me know.

I'll let you close this issue if you think this issue is resolved like this.
